landlab/components/flow_dir_acc_pf/flow_DirAcc_pf.py

```diff
@@ -0,0 +1,886 @@
+            recvr_link[props_Pf <= 0] = -1
+
+            # make sure sum of proportions equals 1
+            if flow_metric in PSINGLE_FMs:
+                # update slope; make pointer to slope location first, in order to not break connection to the slope memory location previously established
+                slope_temp = np.divide(
+                    np.subtract(
+                        npm.repmat(
+                            self.grid.at_node["topographic__elevation"].reshape(
+                                self.grid.number_of_nodes, 1
+                            ),
+                            1,
+                            8,
+                        ),
+                        self.grid.at_node["topographic__elevation"][rcvrs],
+                    ),
+                    self.grid.dx
+                    * np.sqrt(self.grid.at_node["SQUARED_length_adjacent"]),
+                )
+                slope_temp[rcvrs == -1] = 0
+
+            else:
+                props_Pf[props_Pf_col0 == -1, 0] = 1
+                props_Pf = props_Pf.astype(np.float64)  # should be float64
+                # Now, make sure sum is 1 in 64 bits
+                props_Pf[props_Pf == -1] = 0
+                rc64_temp = props_Pf / npm.repmat(
+                    np.reshape(props_Pf.sum(axis=1), [props_Pf.shape[0], 1]), 1, 8
+                )
+                props_Pf[props_Pf[:, 0] != 1, :] = rc64_temp[props_Pf[:, 0] != 1, :]
+                props_Pf[props_Pf == 0] = -1
+
+            if hill_flow:
+                self._hill_prps[:] = props_Pf
+                if flow_metric in PSINGLE_FMs:
+                    idx = np.argmax(rcvrs, axis=1)
+                    self._hill_rcvs[:] = rcvrs[np.arange(rcvrs.shape[0]), idx]
+                    self._hill_slope[:] = slope_temp[np.arange(rcvrs.shape[0]), idx]
+                else:
+                    self._hill_rcvs[:] = rcvrs
+                    # update slope; make pointer to slope location first, in order to not break connection to the slope memory location previously established
+                    self._hill_slope[:] = np.divide(
+                        np.subtract(
+                            npm.repmat(
+                                self.grid.at_node["topographic__elevation"].reshape(
+                                    self.grid.number_of_nodes, 1
+                                ),
+                                1,
+                                8,
+                            ),
+                            self.grid.at_node["topographic__elevation"][rcvrs],
+                        ),
+                        self.grid.dx
+                        * np.sqrt(self.grid.at_node["SQUARED_length_adjacent"]),
+                    )
+                    self._hill_slope[rcvrs == -1] = 0
+
+            else:
+
+                if flow_metric in PSINGLE_FMs:
+                    idx = np.argmax(rcvrs, axis=1)
+                    self._prps[:] = props_Pf[np.arange(rcvrs.shape[0]), idx]
```

Comment:
This looks elegant but it is not exactly doing what I want it to do. We have a matrix, say A with of dimensions r*c. Now idx is a vector of dimension r indicating the column I want to extract. The result should be an rX1 matrix. It indeed feels overkill to create an np array going from 1:r, but when I try your suggestion, I somehow end up with an rXr matrix rather than an rX1 matrix ...

Reply:
I see. How about use *range* instead of *np.arange*?
```python
self._prps[:] = props_Pf[range(rcvrs.shape[0]), idx]
```
or
```python
ij_at_max = range(len(rcvrs)), np.argmax(rcvrs, axis=1)
self._prps[:] = props_Pf[ij_at_max]
self._rcvs[:] = rcvrs[ij_at_max]
self._slope[:] = slope_temp[ij_at_max]
self._recvr_link[:] = recvr_link[ij_at_max]
```

Reply:
Great, exactly what we needed!